File: avis-service/src/main/java/com/example/avis_service/entity/Avis.java
package com.example.avis_service.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "avis")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Avis {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long produitId;

    @Column(nullable = false)
    private String auteur;

    @Column(columnDefinition = "TEXT")
    private String commentaire;

    @Column(nullable = false)
    private int note; // 1 à 5
}

File: app_mobile/lib/models/avis.dart
class Avis {
  final int id;
  final String auteur;
  final String commentaire;
  final int note;

  Avis({
    required this.id,
    required this.auteur,
    required this.commentaire,
    required this.note,
  });

  factory Avis.fromJson(Map<String, dynamic> json) {
    return Avis(
      id: json['id'],
      auteur: json['auteur'],
      commentaire: json['commentaire'],
      note: json['note'],
    );
  }
}